packages/synapse-react-client/src/utils/hooks/useImmutableTableQuery/useImmutableTableQuery.ts

```diff
@@ -102,6 +102,70 @@ export type UseImmutableTableQueryOptions = {
 // When changing the query with the debounce option, the amount of time to wait for additional debounced changes before updating the query
 export const DEBOUNCE_DELAY_MS = 750
 
+/**
+ * Utility to synchronize the URL with the current query request. Synchronization only occurs if `shouldDeepLink` is true.
+ * @param shouldDeepLink whether the query should be synchronized with the URL
+ * @param componentIndex the index of the component to synchronize with the URL - essential if multiple components to synchronize are on the page
+ * @param setQuery - callback used to update the current query
+ * @param initQueryRequest - the initial query request passed via props
+ * @param currentQueryRequest - the query request that is currently used
+ */
+function useSynchronizeQueryWithUrl(
```

Comment:
Extracted these 2 hooks from the larger hook, but only one change that I will call out